packages/plugin-ext/src/main/browser/plugin-ext-frontend-module.ts

```diff
@@ -170,7 +176,23 @@ export default new ContainerModule((bind, unbind, isBound, rebind) => {
             if (endpoint[endpoint.length - 1] === '/') {
                 endpoint = endpoint.slice(0, endpoint.length - 1);
             }
-
+            if (typeof electronRemote !== 'undefined' && typeof electronRemote.session.defaultSession !== 'undefined') {
```

Comment:
it can't be done through a interface/optional implementation if electron module is loaded ?

Reply:
@marechal-p Could you extract `WebviewWidgetFactory` and  override it with `ElectronWebviewWidgetFactory`.

You should be able to call super method and then access `endpoint`, i.e.:
```ts
const widget = super.createWidget(identifier);
const endpoint = widget.externalEndpoint;

// set endpoint cookies here

return widget;
```

Otherwise the place picked very well.